nixos/modules/services/web-apps/sftpgo.nix

```diff
@@ -0,0 +1,370 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sftpgo;
+  defaultUser = "sftpgo";
+  settingsFormat = pkgs.formats.json {};
+  configFile = settingsFormat.generate "sftpgo.json" cfg.settings;
+  hasPrivilegedPorts = builtins.any (port: port > 0 && port < 1024) (
+    builtins.catAttrs "port" (cfg.settings.httpd.bindings
+      ++ cfg.settings.ftpd.bindings
+      ++ cfg.settings.sftpd.bindings
+      ++ cfg.settings.webdavd.bindings
+    )
+  );
+in
+{
+  options.services.sftpgo = {
+    enable = mkEnableOption "Enable sftpgo";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.sftpgo;
+      defaultText = literalExpression "pkgs.sftpgo";
+      description = mdDoc ''
+        Which SFTPGo package to use.
+      '';
+    };
+
+    extraArgs = mkOption {
+      type = with types; listOf str;
+      default = [];
+      description = mdDoc ''
```

Comment:
```suggestion
      description = ''
```

Reply:
please don't we plan on making `mdDoc` mandatory for a transitional period to ensure that all docs are converted correctly

Reply:
There is currently no documentation about this which makes it really hard to enforce in PRs and it probably should require a grace period over one release. Also there will definitely be PRs that miss this.

Reply:
our plan was to add that rule to the documentation once all options were converted, add a nice error message to the manual build for cases where it was forgotten, and keep that in place for a major. if things work out 22.11 would release with mdDoc only, 23.05 would remove it again. in unstable we'd want at least a month between adding enforcement and dropping it again, and considering that we still have no agreement on which markdown processor to use for the manual that doesn't seem like a very tight schedule.

Reply:
If we want to start changing this in review process the documentation must be in place first. Otherwise I have nothing to refer people to.

Reply:
we already have a note in the documentation that writing new docs as docbook is discouraged: https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/development/option-declarations.section.md?plain=1#L58-L66. there's no note yet about docbook being disallowed because there's so much docbook left in nixpkgs. if you think there should be more or clearer docs please do tell, this should be communicated as clearly as possible.